### PR TITLE
HIVE-26692: Check for the expected thrift version before compiling

### PR DESCRIPTION
**Pull request by asolimando**
### What changes were proposed in this pull request?

When using the `thriftif` profile to update the thrift-generated files, the thrift version configured is checked against the expected one, and the task fail explicitly if there is a mismatch, instead of trying to compile anyway.

### Why are the changes needed?

Failing fast saves time and it's more explicit due to the clear error message.

### Does this PR introduce _any_ user-facing change?

No.
### How was this patch tested?

Locally by using a correct/incorrect thrift version and running `mvn clean install -Pthriftif -DskipTests -Dthrift.home=/usr/local`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1773,6 +1773,35 @@
       <id>thriftif</id>
       <build>
         <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
+            <version>1.2.1</version>
```
Comment (ayushtkn) [inline pom.xml:1779]: Seems the version is different that the one being used in hive
https://github.com/apache/hive/blob/81909f3ed294a5c095797346d26d1d1917d42c3a/standalone-metastore/pom.xml#L43

Good if we can extract the variable and use the same variable at all places

Comment (asolimando) [reply pom.xml]: Thanks Ayush for noticing this, unfortunately the `standalone-metastore` pom does not inherit from the topmost one, so we can't share it, but I will introduce a variable with the same name and use the same version.

I have tried locally, everything seems good (normal build and thrift generation, OK and KO paths).

**Review by ayushtkn — APPROVED** [review]
LGTM

**Comment by asolimando** [comment]
@ayushtkn tests are finally green, can we merge this please?

**Pull request merged**